Client/Frontend/Browser/TabManagerStore.swift

```diff
@@ -8,10 +8,11 @@ import Shared
 import XCGLogger
 
 private let log = Logger.browserLogger
-class TabManagerStore {
+class TabManagerStore: FeatureFlagsProtocol {
     fileprivate var lockedForReading = false
     fileprivate let imageStore: DiskImageStore?
     fileprivate var fileManager = FileManager.default
+    fileprivate let profile: Profile
```

Comment:
I'm not against adding `Profile` as a dependency here, but I would like to make sure we're only doing it if absolutely necessary.

Reply:
After looking through it a bit, I think we'll need to pass `profile` in for logic in `shouldOpenHome`. We can also pass just `prefs` from profile as a dependency. What do you think?

Reply:
@nishant2718 I'm good with that, lets just pass the only dependency that is needed 👍